lib/libvarnishapi/vxp_lexer.c

```diff
@@ -126,6 +126,8 @@ vxp_Lexer(struct vxp *vxp)
 					q++;
 					if (q == vxp->e)
 						break;
+				} else if (*q == '\n') {
```

Comment:
I am not sure I understand what is happening here. Should it not be possible to include a literal newline inside a quoted string? Agreed it makes for an awkward display, but do we need to specifically disallow it?

Reply:
Here we break out of the loop if we encounter an actual end of line, not a quoted one. Since an individual query should only span a single line, encountering a new line means the same as reaching `vxp->e`: incomplete string.

Maybe this deserves some VTC coverage.

Reply:
I'm not sure I see a problem with allowing literal newlines in quoted strings. And it would be a change of behaviour to suddenly disallow it. POLA I guess says they should be allowed. But I'm fine either way.

Reply:
I see, I initially thought you misunderstood that change. Now I guess we should gather consensus on this specific point. I will mark this conversation as resolved now that I pushed a second iteration of this patch series, we can discuss this again on the new relevant patch.